deploy-service/common/src/main/java/com/pinterest/deployservice/handler/CommonHandler.java

```diff
@@ -152,33 +156,50 @@ String generateMessage(String buildId, EnvironBean envBean, DeployState state, D
         String webLink = deployBoardUrlPrefix + String.format("/env/%s/%s/deploy/",
             envBean.getEnv_name(),
             envBean.getStage_name());
+        
+        BuildTagsManager buildTagsManager = new BuildTagsManagerImpl(this.tagDAO);
```

Comment:
`buildTagsManager` is available in `ServiceContext`. Therefore you don't need `tagDAO` either.

Reply:
And probably make `buildTagsManager` a member of the class.

Reply:
I tried using it that way and the code doesnt work when doing that. It doesnt pick the tagBean. I saw elsewhere in the code also being done this way. Hence did it this way.

Reply:
Can you add the initialization in `TeletraanServiceContext::setupContext`? It's generally a better pattern and it also makes testing easier.